### ISSUE TITLE
onExternalDrag breaks recomposition of Text color

### ISSUE THREAD

**Issue by AlexeyTsvetkov** [JetBrains/compose-multiplatform#3304]
## Description

I wanted to add color indication inside a drag & drop area.
The area's borders, the icon, and the text are all set to have the same color,
which is held by a mutable state.
When the color state is changed in a button's `onClick`, all elements change the color simultaneously.
However, when the color state is changed in `onExternalDrag`, the text reuses the color from the last composition.

## Expected behavior
All elements change the color simultaneously even when `onExternalDrag` is used.

## Affected platforms
- Desktop

## The environment 
- Kotlin version: 1.8.20
- Compose Multiplatform version*: 1.4.0
- OS version: macOS 13.3
- OS architecture: x86_64
- JDK: Amazon Coretto 17.0.5

## Video

https://github.com/JetBrains/compose-multiplatform/assets/654232/d137fb3c-229c-44c7-ab08-38afb4df7892

## To Reproduce

Drag & drop a file or click the button using the following snippet:
```
import androidx.compose.desktop.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.material.Button
import androidx.compose.material.Icon
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.runtime.*
import androidx.compose.ui.*
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application
import androidx.compose.ui.window.rememberWindowState

fun main() = application {
    val windowState = rememberWindowState(width = 600.dp, height = 400.dp)
    Window(onCloseRequest = ::exitApplication, state = windowState) {
        App()
    }
}

@OptIn(ExperimentalComposeUiApi::class)
@Composable
@Preview
fun App() {
    MaterialTheme {
        Box(modifier = Modifier.padding(5.dp).fillMaxSize().background(Color.White)) {
            val colors = listOf(Color.Black, Color.Blue)
            var activeColorIndex by remember { mutableStateOf(0) }
            fun switchActiveColor() {
                activeColorIndex = (activeColorIndex + 1) % colors.size
            }
            val activeColor = colors[activeColorIndex]

            Column {
                Button(onClick = { switchActiveColor() }, modifier = Modifier.padding(5.dp)) {
                    Text("Change box color")
                }

                Box(
                    modifier = Modifier.size(width = 400.dp, height = 300.dp).padding(5.dp).border(width = 2.dp, activeColor).onExternalDrag(
                        onDragStart = { switchActiveColor()  },
                        onDragExit = { switchActiveColor() },
                    )
                ) {
                    Column(modifier = Modifier.align(Alignment.Center)) {
                        DragAndDropDescription(
                            modifier = Modifier.align(Alignment.CenterHorizontally),
                            color = activeColor
                        )
                    }
                }
            }

        }
    }
}


@Composable
fun DragAndDropDescription(modifier: Modifier, color: Color) {
    val modifier = modifier.padding(vertical = 2.dp)
    println("Icon recomposition $color")
    Icon(
        imageVector = Icons.Default.Send,
        modifier = modifier,
        tint = color,
        contentDescription = ""
    )
    println("Text recomposition: $color")
    Text(
        "Drag & drop a file",
        fontSize = 20.sp,
        modifier = modifier,
        color = color
    )
}
```

**Comment by AlexeyTsvetkov** [comment]
Turns out, the bug is fixed in 1.4.1

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3304">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
